chia/data_layer/data_store.py

```diff
@@ -1395,23 +1395,58 @@ async def clean_node_table(self, writer: Optional[aiosqlite.Connection] = None)
         else:
             await writer.execute(query, params)
 
+    async def get_nodes(self, node_hashes: List[bytes32]) -> List[Node]:
+        hashes = ",".join("?" for _ in node_hashes)
+        async with self.db_wrapper.reader() as reader:
+            # TODO: handle SQLITE_MAX_VARIABLE_NUMBER
+            cursor = await reader.execute(
+                f"SELECT * FROM node WHERE hash IN ({hashes})",
+                [*node_hashes],
+            )
+            rows = await cursor.fetchall()
+
+        hash_to_node = {row["hash"]: row_to_node(row=row) for row in rows}
+        for node_hash in node_hashes:
+            if node_hash not in hash_to_node:
+                raise Exception(f"Node not found for hash: {node_hash.hex()}")
```

Comment:
Could do a set difference as well to get the list of missing stuff i think.  Or at least collect them all to report in the error.